packages/core/test/utils.spec.js

```diff
@@ -0,0 +1,309 @@
+const mockArgv = require('mock-argv');
+const mock = require('mock-fs');
+
+const utils = require('../src/utils');
+
+describe('Utils', () => {
+    describe('.lang()', () => {
+        it('returns correct info for scss', () => {
+            const lang = utils.lang('file.scss');
+            expect(lang.name).toEqual('SCSS');
+            expect(lang.mode).toEqual('scss');
+            expect(lang.scope).toEqual('source.scss');
+        });
+
+        it('returns correct info for nunjucks variants', () => {
+            const expected = {
+                name: 'HTML+Django',
+                mode: 'django',
+                scope: 'text.html.django',
+                color: null,
+            };
+            expect(utils.lang('file.nunjucks')).toEqual(expected);
+            expect(utils.lang('file.nunjs')).toEqual(expected);
+            expect(utils.lang('file.nunj')).toEqual(expected);
+            expect(utils.lang('file.nj')).toEqual(expected);
+            expect(utils.lang('file.jinja2')).toEqual(expected);
+            expect(utils.lang('file.j2')).toEqual(expected);
+        });
+
+        it('returns default value for unknown file type', () => {
+            const expected = {
+                name: '.THIS_IS_NOT_A_FILE',
+                mode: 'plaintext',
+                scope: null,
+                color: null,
+            };
+            expect(utils.lang('file.this_is_not_a_file')).toEqual(expected);
+        });
+    });
+
+    describe('.titlize()', () => {
+        it('returns titlized string', () => {
+            expect(utils.titlize('not-a-title')).toEqual('Not A Title');
+        });
+    });
+
+    describe('.slugify()', () => {
+        it('returns slugified string', () => {
+            expect(utils.slugify('Not A Slug')).toEqual('not-a-slug');
```

Comment:
Maybe worth adding a test with some special char that could break an URL like `?`

Reply:
Hmm, actually the function does not do anything with special characters like that. 

It does, however, replace some unique letters with the more common ones, like `expect(utils.slugify('slug-šäöüõ-slug')).toEqual('slug-saouo-slug');`. so I've added a test for that.